### DISCUSSION TITLE
Search by column that does not appear on table

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#1666] [Help]
Is it possible to allow a table to be searched by a column that doesn't actually appear on the table? I tried accomplishing this several ways (details below) but cannot get it to work.

First I tried using the `hidden(true)` method. While this does hide the column, it does not allow it to be searched. Second I tried using the `hiddenFrom('sm')` method, which both hides the column AND allows it to be searched - but it will only hide the column from the breakpoint that you pass into it, so it still appears at anything below `sm`.

One idea that I had was to use the `visibleFrom()` method and create a stupidly large breakpoint that would never actually occur - something like 4,000 pixels wide. While this would work, it's a giant hack.

**Comment by intrepidws** [comment]
I just tried out my idea with `visibleFrom()` and it looks like the acceptable breakpoints are hardcoded, rather than pulled from my `tailwind.config.js` file (meaning I can't reference my custom breakpoints).

**Reply by intrepidws** [reply]
I wonder if it wouldn't make sense to simply take the breakpoint passed into `visibleFrom()` or `hiddenFrom()` and assume that it's a real breakpoint, rather than only accepting `sm`, `md`, `lg`, `xl` or `2xl`?

**Reply by danharrin** [reply]
We can't accept custom breakpoints really, as those classes would not be purged by Tailwind & built in the CSS.

You can actually add other columns to be searched by passing an array of search columns to any column's `searchable()` - so if you wanted to search by email but you only have a name column, try `TextColumn::make('name')->searchable(['name', 'email'])`

**Reply by intrepidws** [reply]
Thanks Dan, much appreciated for the help.
